k8s-operator/apis/v1alpha1/types_connector.go

```diff
@@ -226,4 +226,7 @@ const (
 
 	IngressSvcValid      ConditionType = `TailscaleIngressSvcValid`
 	IngressSvcConfigured ConditionType = `TailscaleIngressSvcConfigured`
+
+	PGTailscaleServiceValid      ConditionType = `TailscaleServiceValid`      // The Tailscale Service for the ProxyGroup is valid.
+	PGTailscaleServiceConfigured ConditionType = `TailscaleServiceConfigured` // At least one of the ProxyGroup's Pods is advertising the Tailscale Service.
```

Comment:
I've not looked at how the implementtion uses these yet, but given that they are user facing and the 'Tailscale Service' is, to some extent, an implementation detail and also only relevant for the API server proxy, could we perhaps make these `KubeAPIServerProxy[Configured|Valid]`?

Reply:
> 	PGTailscaleServiceConfigured ConditionType = `TailscaleServiceConfigured` // At least one of the ProxyGroup's Pods is advertising the Tailscale Service.

Could we not reuse the 'available' condition here btw?

Reply:
These are all great points - I over-indexed on the way svc-for-pg and ingress-for-pg did things. I'll push a little refactoring shortly.

> which makes me think things are good and ready, but then:

Did you test with the latest version? I pushed a commit last night that should have made the ready condition dependent on all the others being happy, but maybe it was buggy (wasn't really a finished patch, just an illustration for further review comments)

Reply:
After some offline discussion, instead decided to just rename these new conditions so that we only have a single controller responsible for any one condition.